src/Marten/Events/Aggregation/CustomProjection.cs

```diff
@@ -376,9 +376,15 @@ async ValueTask<TDoc> ILiveAggregator<TDoc>.BuildAsync(IReadOnlyList<IEvent> eve
         var documentSessionBase = session as DocumentSessionBase ?? (DocumentSessionBase)session.DocumentStore.LightweightSession();
 
         var slice = new EventSlice<TDoc, TId>(default, session, events);
-        await ApplyChangesAsync(documentSessionBase, slice, cancellation).ConfigureAwait(false);
-
-        ApplyMetadata(slice.Aggregate, events.Last());
+        if (Lifecycle == ProjectionLifecycle.Live)
+        {
+            slice.Aggregate = await BuildAsync(session, slice.Aggregate, slice.Events()).ConfigureAwait(false);
+            ApplyMetadata(slice.Aggregate, events.Last());
```

Comment:
ApplyMetadata also happens in ApplyChangesAsync, so no need to do that again in the else